web/src/buddy_data.ts

```diff
@@ -418,7 +418,9 @@ function get_filtered_user_id_list(
     }
 
     const user_ids = filter_user_ids(user_filter_text, base_user_id_list);
-    return user_ids;
+    // Make sure all the participants are in the list, even if they're inactive
+    const user_ids_set = new Set([...user_ids, ...conversation_participants]);
+    return [...user_ids_set];
```

Comment:
You have this at the top mof the file:

` let base_user_id_list = [...conversation_participants];`

But that value always gets overwritten; so I think you want to delete that declaration or otherwise refactor?

Reply:
The overwriting is adding to it, not removing the participants. `base_user_list` is filtered in `filter_user_ids` and loses some of the contents, which is where the participants were getting removed.

But yeah it makes sense to not add participants until the end now that I've made this choice.